### PR TITLE
Generate loop start, end locations in llvm.loop metadata

### PR DESCRIPTION
**Pull request by kiranchandramohan**
This patch from @peterwaller-arm adds the loop start, end locations in llvm.loop metadata. 
The patch also reorganizes the loop metadata generation to not generate duplicates, fixes some bugs.
A few existing testcases are modified since the llvm.loop metadata is always present.

### CODE REVIEW

**Comment by kiranchandramohan** [comment]
Ping @michalpasztamobica. Any idea why the CI is failing?

**Comment by michalpasztamobica** [comment]
This should fix it: https://github.com/flang-compiler/flang/pull/977

**Review by bryanpkc — APPROVED** [review]
This patch builds cleanly in our internal repo and works fine on the test cases I tried.

**Comment by bryanpkc** [comment]
@shivaramaarao Gentle ping.

File: test/f90_correct/bih-head.f90 [hunk 1]
```diff
@@ -0,0 +1,105 @@
+!
+! Copyright (c) 2017, Arm Ltd.  All rights reserved.
```
Comment (kiranchandramohan) [inline test/f90_correct/bih-head.f90:2]: Change license.

**Comment by kiranchandramohan** [comment]
LLVM-11 comes with a new assertion (https://reviews.llvm.org/D79245) that has to be disabled for this change to pass. Will try to pick that change.

**Comment by kiranchandramohan** [comment]
I think this will have to wait till we move to the llvm.access.group metadata.

**Comment by kiranchandramohan** [comment]
Superseded by https://github.com/flang-compiler/flang/pull/1094

**Pull request closed without merging**